scripts/execute_maestro_tests.sh

```diff
@@ -45,11 +44,30 @@ if [ "$RETRY_COUNT" -eq "$MAX_RETRIES" ]; then
     exit 1
 fi
 
-# Create test results folder.
-mkdir -p /tmp/test_results
-
 # install APK.
 adb install $BITRISE_APK_PATH
 
-# Clear and start collecting logs
-maestro test -e APP_ID=com.stripe.android.financialconnections.example --format junit --include-tags=$MAESTRO_TAGS --output maestroReport.xml maestro/financial-connections
\ No newline at end of file
+TEST_DIR_PATH=maestro/financial-connections
+TEST_RESULTS_PATH=/tmp/test_results
+RETRY_COUNT=0
+
+# Create test results folder.
+mkdir -p $TEST_RESULTS_PATH
+
+for TEST_FILE_PATH in "$TEST_DIR_PATH"/*.yaml; do
+   # Check if tags are present in the test file
+    if awk '/tags:/,/---/' "$TEST_FILE_PATH" | grep -q "$MAESTRO_TAGS"; then
+        # Execute Maestro test flow and retries if failed.
+        while [ "$RETRY_COUNT" -lt "$MAX_RETRIES" ]; do
+            maestro test -e APP_ID=com.stripe.android.financialconnections.example --format junit --output $TEST_FILE_PATH.xml "$TEST_FILE_PATH" && break
+            let RETRY_COUNT=RETRY_COUNT+1
+            echo "Maestro test attempt $RETRY_COUNT failed. Retrying..."
+        done
+        if [ "$RETRY_COUNT" -eq "$MAX_RETRIES" ]; then
+            echo "Maestro tests failed after $MAX_RETRIES attempts."
+            exit 1
+        else
+            RETRY_COUNT=0
+        fi
+    fi
+done
```

Comment:
This script runs Maestro tests on each .yaml file in a specified directory (with a given `tag`), retrying a set number of times if a test fails.

Maestro does not (yet) have built in support for retries.